### ISSUE TITLE
Uncaught TypeError: Cannot read property 'replace' of undefined (in loadbalancerservice.js)

### ISSUE THREAD

**Issue by rossigee** [rancher/rancher#7487]
**Rancher Versions:**
Server: 1.3.2
healthcheck:
ipsec:
network-services:
scheduler:
kubernetes (if applicable):

**Docker Version:**

1.2 and 1.3

**OS and where are the hosts located? (cloud, bare metal, etc):**

AWS

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**

single node rancher

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**

catlle

**Steps to Reproduce:**

After upgrade to 1.3.2, login to UI, presented with a blank page.

**Results:** 

Javascript errors in console (see attached screenshots).

![rancher-ui-error1](https://cloud.githubusercontent.com/assets/5194485/22141386/6b384670-df24-11e6-861a-1317fdab039b.png)
![rancher-ui-error-2](https://cloud.githubusercontent.com/assets/5194485/22141387/6b415c7e-df24-11e6-9dee-4c6aca90959f.png)

**Comment by vincent99** [comment]
I can make it not explode there but it's not clear how you ended up with a balancer that has no image in the first place...

**Comment by andrewmunro** [comment]
I just received this issue too after upgrading. I received a message saying my containers needed to be updated. Upon updating, the UI crashes, appears to be a load balancer with no imageUuid...

![image](https://cloud.githubusercontent.com/assets/1107388/22164340/b3a0949c-df4e-11e6-827a-5a7a2725bbf0.png)

**Comment by rossigee** [comment]
@vincent99 - I ended up in this scenario in the same way as per @andrewmunro, after upgrading the containers of my main default environment.

I notice you've marked this as resolved and tagged it for 1.4 release. Is/was it a database issue? Where's the fix? It would be useful for @andrewmunro and I to understand the cause of the issue and potentially determine a fix or work around for our environments while waiting for the release.

**Comment by vincent99** [comment]
@rossigee As I said I made the UI [not explode there](https://github.com/vincent99/ui/commit/2808f61e9e7da85382a8dfd27e5ad329fa4045d2#diff-d171e926f0def004d92c69cd46e72858), so that's all that's being fixed at this point.

But it's still not clear how you're ending up with a balancer service with no image.  @alena1108 ?

**Comment by rossigee** [comment]
@vincent99 - thanks - i see the fix now - it wasn't linked to the ticket previously.

@alena1108 - pls let me know if you need me to check anything in my rancher db, or can help me to identify the rogue record(s) that need fixing - thanks

**Comment by tfiduccia** [comment]
Version - master 1/24
Verified UI fixed.

**Comment by alena1108** [comment]
@rossigee @andrewmunro have you hit this issue when upgrading from 1.2.x to 1.3.2? or was the initial version different. It would also be helpful to know on what version the LB was originally created too (in case it was created in 1.1.x, it might mean some bug during the DB upgrade process for LB)

**Comment by rossigee** [comment]
For me, it was an upgrade from 1.1.4 directly to 1.3.2.

I tried a few weeks ago to upgrade to 1.2.x, but it crashed spectacularly with lots of issues, so I had to revert to a backup. This upgrade to 1.3.2 seems to have gone a lot better and would probably have worked if it weren't for this issue, but I've again had to restore 1.1.4 from backup as no access to the UI is a definite show-stopper for us.

**Comment by alena1108** [comment]
@rossigee thank you for the update, that explains some things.

@vincent99 in 1.1.x, old style LBs didn't have the image UUID set (UI was passing image as null, and we were defaulting image on per individual container basis down the line). The image - and the LB - was supposed to get upgraded to not null value as a part of environmentUpgrade on 1.1.x-1.2.x upgrade (the must-have call to make to upgrade the environment to the new networking microservices + do the environment upgrade). Does UI require this call as a part of 1.1.x to 1.3.x upgrade as well? Or do we present the user with "upgrade Lb service" option on per individual lb service basis? Wonder why LB upgrade to the initial refactored version was skipped (may be due to the crashed upgrade/revert)

**Comment by alena1108** [comment]
@sangeethah on users setup, something went wrong during 1.1.x to 1.2.x upgrade (the rollback might have gotten in a way, or something else broke). As a result of it, the setup ended up having imageUUID=null for lb services, and that used to break the UI in 1.3. The fix for 1.4 will be - handle and offer upgrade for LBs having null image id. To test things, perform following steps:

1) Create regular lb service.
2) modify the DB to have null image uuid value:

```
{"fields":{"scale":1,"launchConfig":{"instanceTriggeredStop":"stop","kind":"container","networkMode":"managed","privileged":false,"publishAllPorts":false,"readOnly":false,"startOnCreate":true,"stdinOpen":false,"tty":false,"vcpu":1,"imageUuid":null,"ports":[],"expose":["100:100/tcp"],"labels":{"io.rancher.container.agent.role":"environmentAdmin","io.rancher.container.create_agent":"true"},"system":false,"version":"0","healthCheck":{"name":null,"responseTimeout":2000,"interval":2000,"healthyThreshold":2,"unhealthyThreshold":3,"requestLine":null,"port":42,"strategy":null,"recreateOnQuorumStrategyConfig":null,"initializingTimeout":null,"reinitializingTimeout":null}},"startOnCreate":true,"assignServiceIpAddress":false,"lbConfig":{"config":null,"certificateIds":[],"stickinessPolicy":null,"portRules":[{"protocol":"http","priority":1,"sourcePort":100,"serviceId":6,"targetPort":100}],"defaultCertificateId":null},"secondaryLaunchConfigs":[],"token":"k44kWXWDrjeiFmVPksCTWbQZpVo6uKgTMFbwMAXy","executionCount":1,"executionPeriodStart":1485813634226}}
```
3) Validate that the UI presents user with the upgrade option, and that the upgrade goes through.

**Comment by soumyalj** [comment]
Tried with v1.4.0-rc6. Created a Loadbalancer and modify the database entry for the LB to have a NULL image as in the above example. The UI showed does show and option to upgrade. The upgrade is successful and LB is functional
